libbeat/outputs/elasticsearch/client.go

```diff
@@ -178,6 +177,25 @@ func eventBulkMeta(index string, event common.MapStr) bulkMeta {
 	return meta
 }
 
+// getIndex returns the full index name
+// Index is either defined in the config as part of the output
+// or can be overload by the event through setting index
+func getIndex(event common.MapStr, index string) string {
+
+	ts := time.Time(event["@timestamp"].(common.Time)).UTC()
+
+	// Check if index is ste dynamically
+	if dynamicIndex, ok := event["index"]; ok {
```

Comment:
Following suggestion to move forward:
- Introduce "index" under beat.index as temporary solution
- Introduce event object to store all meta data (index, type, timestamp, ?)

I think event object will be a bigger refactoring but I would like to get the option to configure in earlier as an unofficial feature, to play around with it. If it is not used, everything should work as before.